src/main/java/org/jenkinsci/plugins/gitclient/JGitAPIImpl.java

```diff
@@ -850,14 +873,20 @@ public String getRemoteUrl(String name) throws GitException {
      * @throws hudson.plugins.git.GitException if underlying git operation fails.
      */
     @NonNull
-    public Repository getRepository() throws GitException {
+    private Repository getRepository() throws GitException {
         try {
             return new RepositoryBuilder().setWorkTree(workspace).build();
         } catch (IOException e) {
             throw new GitException(e);
         }
     }
 
+    public <T> T withRepository(RepositoryCallback<T> callable) throws IOException, InterruptedException {
+        try (Repository repo = getRepository()) {
+            return callable.invoke(repo, Jenkins.MasterComputer.localChannel);
```

Comment:
IIUC `withRepository` is being called in this case in an agent JVM? If so, we should not be referring to the `Jenkins` type at all—it is unsafe.

Maybe just pass null? I cannot find any `RepositoryCallback` implementations that actually use the `channel` parameter anyway. (And any such attempt would likely be a SECURITY-144 violation!) Maybe we could deprecate it, and switch to a

```java
T invoke(Repository repo) throws IOException, InterruptedException;
```

overload, with a `Util.isOverridden` fallback for compatibility.